ctapipe/image/muon/muon_ring_finder.py

```diff
@@ -1,71 +1,50 @@
 import numpy as np
 import astropy.units as u
-from ctapipe.image.muon.ring_fitter import RingFitter
+from ctapipe.core import Component
 from ctapipe.io.containers import MuonRingParameter
-
-__all__ = ['ChaudhuriKunduRingFitter']
-
-
-class ChaudhuriKunduRingFitter(RingFitter):
-
-    @u.quantity_input
-    def fit(self, x: u.deg, y: u.deg, weight, times=None):
-        """Fast and reliable analytical circle fitting method previously used
-        in the H.E.S.S.  experiment for muon identification
-
-        Implementation based on [chaudhuri93]_
-
-        Parameters
-        ----------
-        x: ndarray 
-            X position of pixel
-        y: ndarray
-            Y position of pixel
-        weight: ndarray
-            weighting of pixel in fit 
-
-        Returns
-        -------
-        X position, Y position, radius, orientation and inclination of circle
+from .fitting import kundu_chaudhuri_circle_fit, taubin_circle_fit
+import traitlets as traits
+
+# the fit methods do not expose the same interface, so we
+# force the same interface onto them, here.
+# we also modify their names slightly, since the names are
+# exposed to the user via the string traitlet `fit_method`
+def kundu_chaudhuri(x, y, weights, mask):
+    '''kundu_chaudhuri_circle_fit with x, y, weights, mask interface'''
+    weights = weights * mask
+    return kundu_chaudhuri_circle_fit(x, y, weights)
+
+def taubin(x, y, weights, mask):
+    '''taubin_circle_fit with x, y, weights, mask interface'''
+    return taubin_circle_fit(x, y, mask)
+
+FIT_METHODS = [kundu_chaudhuri, taubin]
+FIT_METHOD_BY_NAME = {m.__name__: m for m in FIT_METHODS}
+FIT_METHOD_NAMES = list(FIT_METHOD_BY_NAME.keys())
```

Comment:
Do we really need those 3? The dictionary should suffice, right?

Reply:
sure .. I can put the list into the dict comprehension .. and call `list(dict.keys())` twice in lines 31,32 ... I do not think it is nicer ... do you?

Reply:
well .. you had not made the comment if you didn't.. fixing it.

Reply:
done

Reply:
Just seemed like triple bookeeping to me, did not think about the changes further down

Reply:
done